hedera-node/src/main/java/com/hedera/services/contracts/sources/TxnAwareEvmSigsVerifier.java

```diff
@@ -21,131 +21,131 @@
 import com.hedera.services.ledger.accounts.ContractAliases;
 import com.hedera.services.legacy.core.jproto.JKey;
 import com.hedera.services.state.merkle.MerkleAccount;
 import com.hedera.services.state.merkle.MerkleToken;
-import com.hedera.services.store.models.Id;
+import com.hedera.services.store.contracts.WorldLedgers;
 import com.hedera.services.utils.EntityIdUtils;
-import com.hedera.services.utils.EntityNum;
 import com.hederahashgraph.api.proto.java.AccountID;
 import com.swirlds.common.crypto.TransactionSignature;
-import com.swirlds.merkle.map.MerkleMap;
 import org.hyperledger.besu.datatypes.Address;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
 import java.util.Optional;
 import java.util.function.BiPredicate;
-import java.util.function.Supplier;
 
 import static com.hedera.services.exceptions.ValidationUtils.validateTrue;
-import static com.hedera.services.utils.EntityNum.fromAccountId;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.FAIL_INVALID;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_ACCOUNT_ID;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TOKEN_ID;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_HAS_NO_SUPPLY_KEY;
 
 @Singleton
-public class TxnAwareSoliditySigsVerifier implements SoliditySigsVerifier {
+public class TxnAwareEvmSigsVerifier implements EvmSigsVerifier {
 	private final ActivationTest activationTest;
 	private final TransactionContext txnCtx;
 	private final BiPredicate<JKey, TransactionSignature> cryptoValidity;
-	private final Supplier<MerkleMap<EntityNum, MerkleToken>> tokens;
-	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts;
 
 	@Inject
-	public TxnAwareSoliditySigsVerifier(
+	public TxnAwareEvmSigsVerifier(
 			final ActivationTest activationTest,
 			final TransactionContext txnCtx,
-			final Supplier<MerkleMap<EntityNum, MerkleToken>> tokens,
-			final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts,
 			final BiPredicate<JKey, TransactionSignature> cryptoValidity
 	) {
 		this.txnCtx = txnCtx;
-		this.tokens = tokens;
-		this.accounts = accounts;
 		this.activationTest = activationTest;
 		this.cryptoValidity = cryptoValidity;
 	}
 
 	@Override
 	public boolean hasActiveKey(
+			final boolean isDelegateCall,
 			final Address accountAddress,
-			final Address recipient,
-			final Address contract,
 			final Address activeContract,
-			final ContractAliases aliases
+			final WorldLedgers worldLedgers
 	) {
+		if (worldLedgers == null) {
+			throw new InvalidTransactionException("WorldLedgers are not set.", FAIL_INVALID);
+		}
+
 		final var accountId = EntityIdUtils.accountIdFromEvmAddress(accountAddress);
-		final var simpleId = Id.fromGrpcAccount(accountId);
-		final var entityNum = simpleId.asEntityNum();
-		final var account = accounts.get().get(entityNum);
-		validateTrue(account != null, INVALID_ACCOUNT_ID);
+		final var account =
```

Comment:
I don't think `WorldLedgers.accounts()` can ever return `null`, so I'd probably just remove this `null` check too.